src/zm_signal.cpp

```diff
@@ -66,13 +66,13 @@ RETSIGTYPE zm_die_handler(int signal)
 		ucontext_t *uc = (ucontext_t *) context;
 		cr2 = info->si_addr;
 #if defined(__x86_64__)
-	#ifdef __FreeBSD_kernel__
-		ip = (void *)(uc->uc_mcontext.mc_rip);
+	#if defined (__FreeBSD_kernel__) || defined (__MACH__)
+		//ip = (void *)(uc->uc_mcontext->mc_rip);
```

Comment:
Um... this doesn't look right.  You added the MACH test, but remarked out the code... this would fail the FreeBSD case